api/goodies.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+)
+
+// cPARAM is the default string formatter parameter.
+const cPARAM string = "{}"
+
+// -----------------
+// Logger Goodies 🥲
+// -----------------
+
+// formatLog simplifies the use of a string formatter.
+func formatLog(message string, ps ...interface{}) string {
+	pl := len(ps)
+	arr := strings.Split(message, cPARAM)
+	al := len(arr)
+	buf := []string{}
+	i := 0
+	for key := range arr {
+		buf = append(buf, arr[key])
+		if key < pl {
+			buf = append(buf, fmt.Sprintf("%+v", ps[key]))
+			// I add '{}' only if has less parameters
+		} else if pl < (al - 1) {
+			buf = append(buf, cPARAM)
+		}
+		i++
+	}
+	return strings.Join(buf, "")
+}
+
+// --------------------
+// Http Tmux Goodies 😜
+// --------------------
+
+// statusWriter wraps an http response.
+type statusWriter struct {
+	http.ResponseWriter
+	status int
+	length int
+}
+
+// WriteHeader wraps the header writer
+func (w *statusWriter) WriteHeader(status int) {
+	w.status = status
+	w.ResponseWriter.WriteHeader(status)
+}
+
+// WriteHeader wraps the response writer
+func (w *statusWriter) Write(b []byte) (int, error) {
+	if w.status == 0 {
+		w.status = 200
+	}
+	n, err := w.ResponseWriter.Write(b)
+	w.length += n
+	return n, err
+}
+
```

Comment:
to be reviewed

Reply:
@esgn As you wish. I will make the change further to return the correct response status code on every service call. That's not currently the case.

Reply:
Maybe you can explain a bit more that's the point of this code? It'll help us a lot to understand it and be able to give a much more accurate review :wink: Thanks